src/sentry/db/models/paranoia.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import absolute_import
+
+from django.db import models
+from django.db.models import Manager
+from django.db.models.query import QuerySet
+from django.utils import timezone
+
+from sentry.db.models import Model
+
+
+class ParanoidQuerySet(QuerySet):
+    """
+    Prevents objects from being hard-deleted. Instead, sets the
+    ``date_deleted``, effectively soft-deleting the object.
+    """
+
+    def delete(self):
+        self.update(date_deleted=timezone.now())
+
+
+class ParanoidManager(Manager):
+    """
+    Only exposes objects that have NOT been soft-deleted.
+    """
+
+    def get_queryset(self):
+        return ParanoidQuerySet(self.model, using=self._db).filter(
```

Comment:
This should be fine, but we'll want to keep an eye on this without an index. I don't anticipate it being a problem, just something that we're doing implicitly on all queries to this model.